src/UserOuput/UserOutput.js

```diff
@@ -1,11 +1,26 @@
 import React from 'react';
-import './UserOutput.css'
+import './UserOutput.css';
+import ValidationComponent from '../ValidationComponent/ValidationComponent';
+import CharComponent from '../CharComponent/CharComponent';
 
 function userOutput(props) {
+  let letters = null;
+
+  if (props.text.length) {
+    letters = props.text.split('').map((letter, index) =>
```

Comment:
Isn't showing of "used charecter" only one time part of assigmrnt? :)

If not, it is a additional assigment for you :)

'Anton Anton Anton'
should show only 
'Anton'

Reply:
That wasn't in the assignment :) but it's a really interesting task, thanks for giving it to me)

Made changes:
In UserOutput.js
```
if (props.text.length) {
    const uniqueLetters = new Set(text.toLowerCase().replace(/[^a-z]/g, ''));
    uniqueLetters.forEach((letter) =>
      letters.push(<CharComponent
        key={letter}
        letter={letter}
        onLetterClick={() => props.onLetterClick(letter)}/>
      )
    );
  }
```
In App.js:
```
handleLetterDeletion = (letter) => {
    const re = new RegExp(letter, 'gi');
    let text = this.state.text.slice();
    text = text.replace(re, '');

    this.setState({text: text});
}
```